cmd/container-disk-v2alpha/main.c

```diff
@@ -0,0 +1,139 @@
+#include <getopt.h>
+#include <libgen.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include <stdbool.h>
+#include <sys/socket.h>
+#include <sys/stat.h>
+#include <sys/types.h>
+#include <sys/un.h>
+#include <unistd.h>
+#include <signal.h>
+
+#define LISTEN_BACKLOG 50
+#define READINESS_PROBE_FILE "/healthy"
+
+char *copy_path = NULL;
+
+void sig_handler(int signo) {
+    if (copy_path != NULL) {
+        unlink(copy_path);
+    }
+    exit(0);
+}
+
+static void *socket_check(void *arg) {
+    struct stat st = {0};
+    char *copy_path;
+    copy_path = (char *)arg;
+
+    for (;;) {
+        sleep(1);
+        if (stat(copy_path, &st) == -1) {
+            printf("socket %s does not exist anymore\n", copy_path);
+            exit(0);
+        }
+    }
+}
+
+int main(int argc, char **argv) {
+    char *copy_path_dir;
+    bool health_check = false;
+
+    if (signal(SIGTERM, sig_handler) == SIG_ERR) {
+        printf("failed to register SIGTERM callback");
+        exit(1);
+    }
+
+    int c;
+    while(1) {
+        static struct option long_options[] = {
+            /* These options don’t set a flag.
+                We distinguish them by their indices. */
+            {"copy-path",    required_argument, 0, 'c'},
+            {"health-check", no_argument,       0, 'p'},
+            {0, 0, 0, 0}
+        };
+        /* getopt_long stores the option index here. */
+        int option_index = 0;
+        c = getopt_long(argc, argv, "c:p", long_options, &option_index);
+
+        /* Detect the end of the options. */
+        if (c == -1) {
+            break;
+        }
+
+        switch (c) {
+            case 'c':
+                copy_path = optarg;
+                copy_path_dir = strdup(copy_path);
+                dirname(copy_path_dir);
```

Comment:
I am not sure, bit I think that this does nothing?

Reply:
done